src/Google/Ads/GoogleAds/Lib/OAuth2TokenBuilder.php

```diff
@@ -122,14 +169,19 @@ public function build()
         $this->defaultOptionals();
         $this->validate();
 
-        return new UserRefreshCredentials(
-            null,
-            [
+        if ($this->jsonKeyFilePath !== null) {
```

Comment:
Shouldn't we check for both the `jsonKeyFilePath` and the `scopes` here, since they are both required for service accounts?

Reply:
It's already done in validate();
If that requirement fails, it will not go until this line.
What we need is just a simple check what mode a user is using.
We could define a boolean, a string that indicates that during we check in validate() too, but it's obviously overkill.

Reply:
Got it, makes sense.